cli/src/lib.rs

```diff
@@ -201,19 +194,21 @@ impl Iroha {
     #[allow(clippy::too_many_lines)]
     #[iroha_logger::log(name = "init", skip_all)] // This is actually easier to understand as a linear sequence of init statements.
     pub async fn new(
-        config: Configuration,
+        config: Config,
         genesis: Option<GenesisNetwork>,
         logger: LoggerHandle,
     ) -> Result<Self> {
-        let listen_addr = config.torii.p2p_addr.clone();
-        let network = IrohaNetwork::start(listen_addr, config.sumeragi.key_pair.clone())
-            .await
-            .wrap_err("Unable to start P2P-network")?;
+        let network = IrohaNetwork::start(
+            config.iroha.p2p_address.clone(),
+            config.iroha.key_pair.clone(),
+        )
+        .await
+        .wrap_err("Unable to start P2P-network")?;
 
         let (events_sender, _) = broadcast::channel(10000);
         let world = World::with(
-            [genesis_domain(config.genesis.public_key.clone())],
-            config.sumeragi.trusted_peers.peers.clone(),
+            [genesis_domain(config.genesis.public_key().clone())],
```

Comment:
although, every struct field that leaks to user we should make sure is accessible only through a method. We might want to take the `model` macro from the `data_model` which makes sure all the fields are private

Reply:
I don't think it is necessary to hide internal fields of the configuration structures. It's fine if users want to access everything inside. There are few possible invariant violations that might happen, e.g. if they remove peer's self id from `sumeragi.trusted_peers`.

As for "why is this a method access": because `Genesis` is a enum with `Partial` and `Full` variants. Both of them have a public key inside, and `public_key()` method unifies its borrowing.

Reply:
> I don't think it is necessary to hide internal fields of the configuration structures

the reason why we started doing that in the first place is so that we can have the same API in both `iroha_client` and `iroha_smart_contract`.  Since config is used in both the same applies here. Because we're planning to transition to using FFI calls (dynamic linking) in smart contracts some of the fields may be replaced with opaque pointers. You can't do this seemlessly unless you hide internal fields of structures and provide getters.

Reply:
TLDR; we want getters/setters so that we can do internal computation inside

Reply:
just to mention this is relevant for config fields that are accessible from `iroha_client` and `iroha_smart_contract`. Config fields that are only used in `iroha_core` can be defined with which ever API you want

Reply:
Why would `iroha_client` or `iroha_smart_contract` ever want to access config?

Reply:
> Why would iroha_client or iroha_smart_contract ever want to access config?

indeed, although there is `ClientConfiguration` for `iroha_client`?

Reply:
Hm, I don't see any usage of configuration in `iroha_swarm_contract`. I don't understand how the `iroha_client::config::Config` might be used in FFI in future. 
Currently, all access to internal fields of the config is performed from tests and `iroha_client_cli` and it seems fine without getters/setters.